### PR TITLE
fix(external docs): Display `encoding.codec` options

### LINKED ISSUE #8530: `encoding.codec` options do not appear to be rendering on site
Example: https://vector.dev/docs/reference/configuration/sinks/aws_kinesis_firehose/#encoding.codec

Screen Shot 2021-07-30 at 12 26 01 PM

### PR DESCRIPTION
**Pull request by DavidWeid**
Fixes #8530

### CODE REVIEW

File: website/cue/reference/components/sinks.cue [hunk 1]
```diff
@@ -163,6 +163,10 @@ components: sinks: [Name=string]: {
 								required:    true
 								type: string: {
 									examples: features.send.encoding.codec.enum
+									enum: {
+										json: "Array of JSON objects, which must be a JSON array containing JSON objects."
+										text: "Newline-delimited text, with each line forming a message."
+										}
```
Comment (dre-sctg) [inline website/cue/reference/components/sinks.cue:169]: you might be missing few here, https://github.com/timberio/vector/issues/8530#issuecomment-916947401 mentions also `ndjson` and `logfmt`.

Comment (dre-sctg) [reply website/cue/reference/components/sinks.cue:169]: plz note that I've never done any Cue, so I honestly don't (~have any clue~) understand how those values are interpolated around the code. I would imagine that there's some some magic that filters them over the list in the actual component (https://github.com/timberio/vector/blob/6c94aad0f55b8e50b624f50c9d656f26637c2bb9/website/cue/reference/components/sinks/aws_s3.cue#L33-L36).

Comment (DavidWeid) [reply website/cue/reference/components/sinks.cue:169]: @dre-sctg Thanks - it looks like this components/sinks.cue file would be where the generic usage would go, i.e. set enum to `features.send.encoding.codec`, which would be set in the appropriate `.cue` file: website/cue/reference/components/sinks/aws_kinesis_firehose.cue, for example (which lists enum options as json and text) or website/cue/referrence/components/sinks/aws_s3.cue (which lists enum options as ndjson or text).

Will need to update the layouts/partials/data.html file to update the HTML rendering of the enum options. Lots of files linked it seems 🤔

**Comment by jszwedko** [comment]
@DavidWeid apologies! I didn't realize this was a cue data issue. I think I resolved it in https://github.com/timberio/vector/pull/9109. You can feel free to punt any issues related to the cue data back to us. I know your focus is more on the website side of things.

**Comment by DavidWeid** [comment]
@jszwedko Awesome, good to hear. Just took me a while to identify what the issue was exactly, but thanks! Cue is something else 😆

**Pull request closed without merging**